### PR TITLE
Add api key when invoking orchard; minor refactor;

### PR DESCRIPTION
**Pull request by damianxu88**
Local testing successful end to end, from manually invoking Sprinkler control -> scheduler calls Orchard QA -> EMR/EC2 job created and executed in QA

### CODE REVIEW

File: orchard/orchard_runner.go [hunk 1]
```diff
@@ -79,20 +79,22 @@ func s3ArtifactGenerate(artifact string, command string) (string, error) {
...
 	if len(cmds) < 1 {
 		return "", fmt.Errorf("Invalid command line %s", command)
 	}
-	out, err := exec.Command(cmds[0], cmds[1:]...).Output()
+	cmd := exec.Command(cmds[0], cmds[1:]...)
+	out, err := cmd.CombinedOutput()
+	output := string(out)
 	if err != nil {
-		return "", fmt.Errorf("exec command %v has error:%w", command, err)
+		return "", fmt.Errorf("exec command %v has error: %w: %s", command, err, output)
```
Comment (damianxu88) [inline orchard/orchard_runner.go:95]: this is to throw the real error along with `exit status 1`

File: orchard/orchard_runner.go [hunk 2]
```diff
@@ -79,20 +79,22 @@ func s3ArtifactGenerate(artifact string, command string) (string, error) {
...
-		return "", fmt.Errorf("parse command line error%w", err)
+		return "", fmt.Errorf("parse command line error %w", err)
 	}
 	if len(cmds) < 1 {
 		return "", fmt.Errorf("Invalid command line %s", command)
 	}
-	out, err := exec.Command(cmds[0], cmds[1:]...).Output()
+	cmd := exec.Command(cmds[0], cmds[1:]...)
+	out, err := cmd.CombinedOutput()
+	output := string(out)
```
Comment (schowsf) [inline orchard/orchard_runner.go:93]: maybe swap L93 with the err check.  If err, not call `string(out)`

Comment (damianxu88) [reply orchard/orchard_runner.go:93]: @schowsf the `string(out)` is now needed no matter error or not, when error, it contains the error stack, if not, it contains the json payload output

**Review by schowsf — APPROVED** [review]

File: cmd/scheduler.go [hunk 3]
```diff
@@ -57,9 +63,23 @@ func init() {
...
+		"orchardAddress",
 		"http://ws:8081",
 		"address to orchard service",
 	)
-	viper.BindPFlag("scheduler.orchardAddress", schedulerCmd.Flags().Lookup("orchard"))
+	viper.BindPFlag("scheduler.orchard.address", schedulerCmd.Flags().Lookup("orchard"))
+
+	schedulerCmd.Flags().String(
+		"orchardAPIKeyName",
+		"x-api-key",
```
Comment (realstraw) [inline cmd/scheduler.go:74]: this needs to be empty

Comment (damianxu88) [reply cmd/scheduler.go]: @realstraw changed

**Review by realstraw — APPROVED** [review]

**Pull request merged**